Tasks/AzureFileCopy/AzureUtilityGTE1.1.0.ps1

```diff
@@ -0,0 +1,32 @@
+# This file implements IAzureUtility for Azure PowerShell version >= 1.1.0
+
+Import-Module ./AzureUtilityGTE1.0.ps1 -Force
+
+function Set-AzureMachineCustomScriptExtension
+{
+    param([string]$resourceGroupName,
+          [string]$vmName,
+          [string]$name,
+          [string[]]$fileUri,
+          [string]$run,
+          [string]$argument,
+          [string]$location)
+
+    if(-not [string]::IsNullOrEmpty($resourceGroupName) -and -not [string]::IsNullOrEmpty($vmName) -and -not [string]::IsNullOrEmpty($name))
+    {
+        Write-Host (Get-LocalizedString -Key "[Azure Call]Setting the custom script extension '{0}' for vm '{1}'" -ArgumentList $name, $vmName)
+		Write-Verbose -Verbose "Set-AzureRmVMCustomScriptExtension -ResourceGroupName $resourceGroupName -VMName $vmName -Name $name -FileUri $fileUri  -Run $run -Argument $argument -Location $location -ErrorAction Stop -Verbose"
+        $result = Set-AzureRmVMCustomScriptExtension -ResourceGroupName $resourceGroupName -VMName $vmName -Name $name -FileUri $fileUri  -Run $run -Argument $argument -Location $location -ErrorAction Stop -Verbose		
+        Write-Host (Get-LocalizedString -Key "[Azure Call]Set the custom script extension '{0}' for vm '{1}'" -ArgumentList $name, $vmName)
+		if($result.IsSuccessStatusCode -eq $true)
```

Comment:
Can you also add a condition, if ($result.IsSuccessStatusCode -eq $false) and set the status and error or exception value if it meets.